src/garage/_dtypes.py

```diff
@@ -559,3 +559,214 @@ def output_space(self):
 
         """
         return self._output_space
+
+
+class SampleBatch(
```

Comment:
We need to figure out naming here, since this is the batch version of `TimeStep` above. Either this is a `TimeStepBatch` or the dtype above is called `Sample`

I favor `TimeStepBatch` because sample is an overloaded term (as is `Step`).